libbpf-rs/tests/bin/src/stream.bpf.c

```diff
@@ -0,0 +1,65 @@
+#include <bpf/bpf_helpers.h>
+
+/* Definition of can_loop taken from bpf_experimental.h. */
+#ifdef __BPF_FEATURE_MAY_GOTO
+#define can_loop                                                               \
+    ({                                                                         \
+        __label__ l_break, l_continue;                                         \
+        bool ret = true;                                                       \
+        asm volatile goto("may_goto %l[l_break]" :: ::l_break);                \
+        goto l_continue;                                                       \
+    l_break:                                                                   \
+        ret = false;                                                           \
+    l_continue:;                                                               \
+        ret;                                                                   \
+    })
+#else
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#define can_loop                                                               \
+    ({                                                                         \
+        __label__ l_break, l_continue;                                         \
+        bool ret = true;                                                       \
+        asm volatile goto("1:.byte 0xe5;		\
+		      .byte 0;				\
+		      .long ((%l[l_break] - 1b - 8) / 8) & 0xffff;	\
+		      .short 0" :: ::l_break);                                 \
+        goto l_continue;                                                       \
+    l_break:                                                                   \
+        ret = false;                                                           \
+    l_continue:;                                                               \
+        ret;                                                                   \
+    })
+#else
+#define can_loop                                                               \
+    ({                                                                         \
+        __label__ l_break, l_continue;                                         \
+        bool ret = true;                                                       \
+        asm volatile goto("1:.byte 0xe5;		\
+		      .byte 0;				\
+		      .long (((%l[l_break] - 1b - 8) / 8) & 0xffff) << 16;	\
+		      .short 0" :: ::l_break);                                 \
+        goto l_continue;                                                       \
+    l_break:                                                                   \
+        ret = false;                                                           \
+    l_continue:;                                                               \
+        ret;                                                                   \
+    })
+#endif
+#endif
+
+volatile u64 i;
+
+/* Trigger a may_goto timeout to emit a streams error. */
```

Comment:
So this `may_goto` timeout is a way to force the kernel to emit data to stderr, is that how it works? And there is no way to write to a stream directly? And what are the requirements for `may_goto` to be available? Is it just a certain LLVM version? Or kernel config? It's also not exactly clear what happens if it's not available. Will the test still work?

If there is no easier way to trigger a stream write, can you please expand the comment a bit to make it clear who does what and in response to what exactly?